lib/resty/core/shdict.lua

```diff
@@ -387,6 +403,39 @@ local function shdict_flush_all(zone)
 end
 
 
+local function shdict_get_stats(zone, detail)
+    zone = check_zone(zone)
+
+    C.ngx_http_lua_ffi_shdict_get_stats_slot_count(zone, slot_count_buf)
+    local stats = ffi_new("ngx_slab_stat_t[?]", slot_count_buf[0])
+    C.ngx_http_lua_ffi_shdict_get_stats_slots_and_min_size(zone, slot_count_buf[0], stats, slot_min_size_buf)
+    local slot_count = tonumber(slot_count_buf[0])
+    local slot_size = tonumber(slot_min_size_buf[0])
+    local total_used = 0
+    local slots = {}
```

Comment:
Again, can we allow the caller to recycle the Lua table here? This way we can avoid allocating a new Lua table in every invocation of this Lua function.